recorder.hpp

```diff
@@ -63,6 +63,14 @@ class Recorder {
         double longitude,
         double horizontalUncertainty,
         double altitude) = 0;
+
+    #ifdef USE_OPENCV_VIDEO_RECORDING
+    // Returns reused cv::Mat pointers to be used to store image data for  addFrame(Group) to avoid
+    // reallocating memory. Returns an empty vector, if not enough free frames are available and
+    // adds a dropped frame event to JSONL output file.
+    virtual std::vector<cv::Mat> getEmptyFrames(size_t number, double time, int width, int height, int type) = 0;
```

Comment:
Prefer this to avoid unnecessary heap allocations:
```suggestion
    virtual void getEmptyFrames(size_t number, double time, int width, int height, int type, std::vector<cv::Mat> &out) = 0;
```
If I understood coorectly, this is used on each frame and `number` is the number of cameras (1 or 2)

Reply:
Good point, fixed